scripts/common/primer_autodetection.py

```diff
@@ -79,52 +78,44 @@ def build_primers_automaton(primer_index: Path) -> dict[str, PrimerAutomaton]:
     return primers_automaton
 
 
-def count_primer_matches(sample_fastq: Path, primers_automaton: dict) -> dict:
+def count_primer_matches(sample_fastq: Path, primers_automaton: dict, primer_base_window: int) -> dict:
     """
     Perform an exact search of the primer sequences (all schemes) in the sample_fastq
     """
     # Structure for storing the unique primer sequences
     unique_hits: dict[str, set] = {p: set() for p in primers_automaton}
 
+    n_reads = 0
+
     # look up the primer sequences in the automaton
     # reading the sample fastq once for all
     with gzip.open(sample_fastq, "rt") as fastq_file:
         # for each sample read, it counts the number of primers found for each primer scheme.
         # This search is performed using an ahocorasick structure and optimised to search
         # at the beginning of the read only
         for sample_sequence in SeqIO.parse(fastq_file, "fastq"):
-            sample_read = str(sample_sequence.seq)
+            # fastq_file is assumed to be adapter-trimmed, e.g.
+            # * trim_galore (illumina)
+            # * porechop (ont)
+            # Trimmer algorithms might not remove the adapter sequences entirely.
+            # Therefore, here the primer is searched in a prefix window of primer_base_window bases,
+            # rather than at the very beginning of the read.
+            sample_read = str(sample_sequence.seq)[0:primer_base_window]
```

Comment:
Search primers in this window only (basically not the whole read) for sensitivity and performance reasons.